siwe_auth/backend.py

```diff
@@ -44,18 +42,7 @@ class SiweBackend(BaseBackend):
     Authenticate an Ethereum address as per Sign-In with Ethereum (EIP-4361).
     """
 
-    def authenticate(self, request, signature: str = None, siwe_message: SiweMessage = None):
-        body = json.loads(request.body)
-
-        if siwe_message is None:
-            siwe_message = SiweMessage(
-                message={
-                    re.sub(r"(?<!^)(?=[A-Z])", "_", k).lower(): v
-                    for k, v in body["message"].items()
-                }
-            )
-            signature = body["signature"]
-
+    def authenticate(self, request, signature: str, siwe_message: SiweMessage):
```

Comment:
Can we remove `request` from this function? (I'm briefly skimming on a phone and I didn't see any uses of it)

Reply:
Unfortunately, we cannot. This is baked in to the BaseBackend https://github.com/django/django/blob/282d58e19385ee4e5c125a9d3cba820949314f3f/django/contrib/auth/backends.py#L13

Reply:
Oh! Good to know